src/components/BottomSheet.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { useMemo, useRef, useCallback, useState } from 'react';
+import { Text, TouchableOpacity, StyleSheet, View } from 'react-native';
+import colors from '@styles/colors';
+import BottomSheet, { BottomSheetScrollView } from '@gorhom/bottom-sheet';
+import { GestureHandlerRootView } from 'react-native-gesture-handler';
+import { WeatherComponent } from '@components/WeatherComponent';
+import { ExtraInfoType } from '@components/ExtraInfo';
+import { WarningType } from '@components/Warning';
+import { WeatherType } from '@components/WeatherIcon';
+import { MAIN_HORIZONTAL_OFFSET } from '@styles/constants';
+import { BottomSheetButtons } from '@utils/constants';
+
+export const Sheet = () => {
+    const sheetRef = useRef<BottomSheet>(null);
+
+    const snapPoints = useMemo(() => ['29%', '96%'], []);
+
+    const handleSheetChanges = useCallback((index: number) => {
+        console.log('handleSheetChanges', index);
```

Comment:
It is better to save new index and then if you change the section, set that index to the bottom sheet. Create useRef and pass it to the index prop of the bottom sheet